react/constants/SearchHelpers.js

```diff
@@ -93,28 +93,42 @@ function removeFilter(map, rest, { type, slug, pagesPath }) {
     return { map, rest }
   }
 
+  if (type !== CATEGORIES_TYPE) {
+    let categoriesFiltered = 0
+
+    const mapWithoutCategories = map.filter(s => {
+      if (categoriesFiltered === categoryCount) {
+        return true
+      }
+
+      categoriesFiltered++
+
+      return s !== categoryMapSymbol
+    }).filter((_, i) => i !== restIndex)
+
+    const lengthDifference = map.length - mapWithoutCategories.length - 1
```

Comment:
What if this result becomes negative?

Reply:
It never will.

The `mapWithoutCategories` is at least 1 unit smaller than the `map` array because I'm removing one filter from the array.